errors_test.go

```diff
@@ -331,3 +331,42 @@ type errorWithNilCause struct{}
 
 func (*errorWithNilCause) Error() string { return "" }
 func (*errorWithNilCause) Cause() error  { return nil }
+
+func TestLookupTag(t *testing.T) {
+	taggedErr := WithTags(errors.New("tagged"), T("key", "value1"))
+
+	tests := []struct {
+		err    error
+		result string
+	}{
+		{
+			err: nil,
+		},
+		{
+			err: errors.New("foreign"),
+		},
+		{
+			err:    taggedErr,
+			result: "value1",
+		},
+		{
+			err:    WithTags(Wrap(taggedErr, "double"), T("key", "value2")),
+			result: "value2",
+		},
```

Comment:
Maybe add a test with multiple nesting of `WithTags` with the same tag key? That way you can ensure that always the same value is returned (some programs could start relying on the behavior of the package).

Reply:
The last test case does just that. `"key"` is assigned twice with `"value1"` and `"value2"`.
The case ensures that the func returns `"value2"`, but now I'm not sure that would be the desired behavior.

Reply:
Oh you're right! I missed `taggedErr`.

I don't think there's a _right_ behavior, we just need to have one and stick to it. Returning the tag from the last wrapped error seems intuitive.

Looking good 👍